### PR TITLE
feat: add project management functionality with CRUD operations and role/member management

### LINKED ISSUE #34: [Epic] Project Management
## Summary
Foundational workspace management for organizing independent streams of work.

## Problem
Users need to isolate data and workflows into logical projects to maintain organization and security boundaries.

## Scope
- Project CRUD operations.
- Project metadata (Titles, Descriptions, Status).
- Project archival and soft-deletion.

### PR DESCRIPTION
**Pull request by pikann**
- [x] Add `ErrNameInvalid` to project domain errors
- [x] Update `project_service.go` to return `ErrNameInvalid` when name is empty after trim
- [x] Add `CodeProjectNameInvalid` to `apierr/codes.go`
- [x] Map `ErrNameInvalid` → `CodeProjectNameInvalid` → 400 in `presenter/response.go`
- [x] Add a test for whitespace-only project name (400 response)
- [x] Add `TestUpdateProject_PreservesOmittedFields` to verify PATCH semantics don't wipe existing description
- [x] Fix `RequireAnyPermissions` to surface scope-resolution errors (400) instead of hiding them behind 403
- [x] Add regression tests for invalid-param case in `RequireAnyPermissions`
- [x] Fix incorrect endpoint path comments in `project_dto.go` (`POST /projects`, `PATCH /projects/:projectId`)
- [x] Fix `toProjectEntity` to return error on UUID parse failure and JSON unmarshal failure
- [x] Fix `toProjectRoleEntity` to return error on UUID parse failure and JSON unmarshal failure
- [x] All tests pass

### CODE REVIEW

File: services/api/internal/service/project/project_service.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+func (s *Service) GetByID(ctx context.Context, id uuid.UUID) (*projectdom.Project, error) {
+	return s.repo.FindByID(ctx, id)
+}
+
+// Create defines and persists a new project.
+func (s *Service) Create(ctx context.Context, in projectdom.CreateProjectInput) (*projectdom.Project, error) {
+	name := strings.TrimSpace(in.Name)
+	if name == "" {
+		return nil, projectdom.ErrNameTaken
+	}
```
[the remark this answers is not in the document]
Comment (pikann) [reply services/api/internal/service/project/project_service.go:59]: @copilot apply changes based on [this feedback](https://github.com/Paca-AI/paca/pull/70#discussion_r3025753701)

File: services/api/internal/transport/http/handler/project_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+	projID := uuid.New()
+	r := newProjectRouter(&mockProjectSvc{
+		update: func(_ context.Context, id uuid.UUID, in projectdom.UpdateProjectInput) (*projectdom.Project, error) {
+			return &projectdom.Project{ID: id, Name: in.Name}, nil
+		},
+	})
+
+	w := do(t, r, http.MethodPatch, fmt.Sprintf("/admin/projects/%s", projID),
+		jsonBody(t, map[string]any{"name": "updated"}))
+	if w.Code != http.StatusOK {
```
[the remark this answers is not in the document]
Comment (pikann) [reply services/api/internal/transport/http/handler/project_handler_test.go:346]: @copilot apply changes based on [this feedback](https://github.com/Paca-AI/paca/pull/70#discussion_r3025753737)

File: services/api/internal/transport/http/middleware/authz.go [hunk 3]
```diff
@@ -81,3 +81,60 @@ func RequirePermissions(authorizer *authz.Authorizer, scope ScopeResolver, permi
...
+		for _, group := range groups {
+			resolver := group.Scope
+			if resolver == nil {
+				resolver = GlobalScope()
+			}
+			projectID, err := resolver(c)
+			if err != nil {
+				// Scope resolution error (e.g. malformed project ID in param): skip group.
+				continue
+			}
```
[the remark this answers is not in the document]
Comment (pikann) [reply services/api/internal/transport/http/middleware/authz.go:131]: @copilot apply changes based on [this feedback](https://github.com/Paca-AI/paca/pull/70#discussion_r3032822557)

File: services/api/internal/transport/http/dto/project_dto.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+// CreateProjectRequest is the body for POST /v1/admin/projects.
+type CreateProjectRequest struct {
+	Name        string         `json:"name" binding:"required"`
+	Description string         `json:"description"`
+	Settings    map[string]any `json:"settings"`
+}
+
+// UpdateProjectRequest is the body for PATCH /v1/admin/projects/:projectId.
+type UpdateProjectRequest struct {
```
[the remark this answers is not in the document]
Comment (pikann) [reply services/api/internal/transport/http/dto/project_dto.go]: @copilot apply changes based on [this feedback](https://github.com/Paca-AI/paca/pull/70#discussion_r3032822595)

File: services/api/internal/repository/postgres/project_repository.go [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+}
+
+// --- Mapping helpers --------------------------------------------------------
+
+func toProjectEntity(rec *projectRecord) (*projectdom.Project, error) {
+	id, _ := uuid.Parse(rec.ID)
+	var settings map[string]any
+	if err := json.Unmarshal(rec.Settings, &settings); err != nil {
+		settings = map[string]any{}
+	}
```
[the remark this answers is not in the document]
Comment (pikann) [reply services/api/internal/repository/postgres/project_repository.go:419]: @copilot apply changes based on [this feedback](https://github.com/Paca-AI/paca/pull/70#discussion_r3032822616)
